--- a/core/src/main/java/com/taobao/arthas/core/shell/term/impl/http/DirectoryBrowser.java
+++ b/core/src/main/java/com/taobao/arthas/core/shell/term/impl/http/DirectoryBrowser.java
@@ -173,11 +173,8 @@
                 if (file.isHidden() || !file.exists() || file.isDirectory() || !file.isFile()) {
                     return null;
                 }
-
-                RandomAccessFile raf;
-                try {
-                    raf = new RandomAccessFile(file, "r");
-                } catch (Exception ignore) {
+                try (RandomAccessFile raf = new RandomAccessFile(file, "r")) {
+                } catch (java.lang.Exception ignore) {
                     return null;
                 }
                 long fileLength = raf.length();
