--- a/MyPerf4J-Base/src/main/java/cn/myperf4j/base/util/concurrent/ExecutorManager.java
+++ b/MyPerf4J-Base/src/main/java/cn/myperf4j/base/util/concurrent/ExecutorManager.java
@@ -31,6 +31,7 @@
                 executorService.awaitTermination(timeout, unit);
             } catch (InterruptedException e) {
                 Logger.error("ExecutorManager.stopAll()", e);
+                Thread.currentThread().interrupt();
             }
         }
     }
