--- a/goodskill-chat-provider/src/main/java/com/goodskill/chat/client/http/ChatHttpClient.java
+++ b/goodskill-chat-provider/src/main/java/com/goodskill/chat/client/http/ChatHttpClient.java
@@ -1,4 +1,7 @@
 package com.goodskill.chat.client.http;
+import DefaultFullHttpRequest;
+import HttpContent;
+import HttpRequest;
 
 import cn.hutool.core.bean.BeanUtil;
 import com.goodskill.chat.client.ChatClient;
@@ -34,6 +37,7 @@
                 channel = initConnection();
             } catch (InterruptedException e) {
                 e.printStackTrace();
+                Thread.currentThread().interrupt();
             }
         }
 
