--- a/src/main/java/me/botsko/prism/database/mysql/HikariHelper.java
+++ b/src/main/java/me/botsko/prism/database/mysql/HikariHelper.java
@@ -40,16 +40,14 @@
                     prop.setProperty("dataSource." + name, val);
                 }
             }
-            try {
-                if (!propFile.getParentFile().exists() && !propFile.getParentFile().mkdirs()) {
-                    Prism.log("Prism Directory couldn't be created");
-                }
-                OutputStream out = new FileOutputStream(propFile);
-                prop.store(out, "Prism Hikari Datasource Properties for"
-                        + " advanced database Configuration");
-                Prism.log("Database Configuration saved to - " + propFile.getPath());
-            } catch (IOException e) {
-                Prism.log("Could not save Hikari.properties - " + e.getMessage());
+        try (OutputStream out = new FileOutputStream(propFile)) {
+            if ((!propFile.getParentFile().exists()) && (!propFile.getParentFile().mkdirs())) {
+                Prism.log("Prism Directory couldn't be created");
             }
+            prop.store(out, "Prism Hikari Datasource Properties for" + " advanced database Configuration");
+            Prism.log("Database Configuration saved to - " + propFile.getPath());
+        } catch (java.io.IOException e) {
+            Prism.log("Could not save Hikari.properties - " + e.getMessage());
+        }
         }
 }
